lib/vanity/experiment/ab_test.rb

```diff
@@ -126,16 +176,23 @@ def fingerprint(alternative)
       def choose(request=nil)
         if @playground.collecting?
           if active?
-            identity = identity()
-            index = connection.ab_showing(@id, identity) || connection.ab_assigned(@id, identity)
-            unless index
-              index = alternative_for(identity).to_i
-              save_assignment_if_valid_visitor(identity, index, request) unless @playground.using_js?
+            if enabled?
+              identity = identity()
```

Comment:
To avoid some of the deep nesting here, could we extract L180-185 to a private helper method?